plugin-aggregation/src/transform.rs

```diff
@@ -0,0 +1,740 @@
+        measurements.clear();
+
+        // Fill it again with the not needed points.
+        measurements.merge(&mut not_needed_measurement_point);
+
+        self.buffer_bouncer(measurements)
+    }
+}
+
+/// Returns true if the vec contains enough data to compute the aggregation
+/// for the configured window.
+fn contains_enough_data(interval: Duration, values: &Vec<MeasurementPoint>, min_timestamp: Timestamp) -> bool {
+    let elapsed = values[values.len() - 1]
+        .timestamp
+        .duration_since(min_timestamp)
+        .unwrap();
+    elapsed >= interval
+}
+
+/// Get the IDs of the first and last measurement point that are
+/// inside the interval window.
+fn get_ids(
+    interval: Duration,
+    values: &Vec<MeasurementPoint>,
+    min_timestamp: Timestamp,
+) -> anyhow::Result<(usize, usize)> {
+    let i: usize = 0;
+
+    let Some(j) = values.iter().position(|point| {
+        let elapsed = point.timestamp.duration_since(min_timestamp).unwrap();
+        elapsed >= interval
+    }) else {
+        return Err(anyhow!("could not compute the IDs for the current sub_vec"));
+    };
+
+    Ok((i, j - 1))
+}
+
+/// Compute the rounded timestamp of reference_timestamp to the closest one
+/// bellow it which is a multiple of interval.
+fn compute_min_timestamp(reference_timestamp: Timestamp, interval: Duration) -> Timestamp {
+    let reference_unix_timestamp = reference_timestamp.to_unix_timestamp();
+
+    let reference_timestamp_duration = Duration::new(reference_unix_timestamp.0, reference_unix_timestamp.1);
+    let reference_timestamp_nanos = reference_timestamp_duration.as_nanos();
+
+    // The formula used is:
+    // reference_timestamp = k * interval + p, (k,p) ∈ N²
+    // => p = reference_timestamp % interval
+    // => new_timestamp = k * interval = reference_timestamp - p
+    let p = reference_timestamp_nanos % interval.as_nanos();
+
+    let new_timestamp_secs = (reference_timestamp_nanos - p) / 1_000_000_000;
+    let new_timestamp_nanosecs = (reference_timestamp_nanos - p) % 1_000_000_000;
+
+    let new_timestamp = UNIX_EPOCH + Duration::new(new_timestamp_secs as u64, new_timestamp_nanosecs as u32);
+    Timestamp::from(new_timestamp)
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use std::time::{Duration, SystemTime, UNIX_EPOCH};
+    use time::{format_description::well_known::Rfc3339, OffsetDateTime};
+
+    use alumet::{
+        measurement::{MeasurementBuffer, MeasurementPoint, Timestamp, WrappedMeasurementValue},
+        metrics::RawMetricId,
+        resources::{Resource, ResourceConsumer},
+    };
+
+    use crate::transform::{compute_min_timestamp, contains_enough_data};
+
+    use super::get_ids;
+
+    /// Parses an RFC 3339 date-and-time string into a Timestamp value.
+    pub(crate) fn timestamp_from_rfc3339(timestamp: &str) -> Timestamp {
+        Into::<SystemTime>::into(OffsetDateTime::parse(timestamp, &Rfc3339).unwrap()).into()
```

Comment:
When you need to specify the result type of the conversion, the idiomatic way is to use the mirror trait `From`:
```suggestion
        SystemTime::from(OffsetDateTime::parse(timestamp, &Rfc3339).unwrap()).into()
```

Into is only a convenience built upon From (implementing from automatically provides into).

Reply:
Okay, I didn't know 😅